---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Model Abstraction and Temporal Behavior Analysis of Genetic Regulatory Networks
subtitle: ''
summary: ''
authors:
- Hiroyuki Kuwahara
tags:
- 'genetic circuit'
- 'stochastic logic'
categories: []
date: '2007-12-01'
lastmod: 2021-01-15T21:34:48Z
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-01-15T21:34:48.606746Z'
publication_types:
- '7'
abstract: With advances in technologies such as high throughput data collection and
  genome sequencing methods, the Human Genome Project which, among other things, determined
  the human DNA sequence and identified all the genes in human DNA was completed at
  least two years ahead of time. As these technologies are becoming more accurate,
  efficient, and cost effective and a massive amount of genomic and proteomic data
  are becoming available at a rapid pace, we are now in the position to face the challenge
  to understand how these genes coupled with environmental stimuli orchestrate the
  regulation of cell-level behaviors. However, understanding such complex systems
  is very expensive and is most likely impractical with wet-lab experiments alone
  as the amount and the complexity of data substantially increase, requiring the integration
  of computational methods to make the process more efficient. To allow for substantial
  acceleration in computational analysis, this dissertation develops a model abstraction
  methodology for biochemical systems which systematically performs various model
  abstractions to reduce the complexity of computational biochemical models. Our methodology
  is particularly useful for systems with small molecular counts that require the
  discrete and stochastic representation and thus demand substantial computational
  requirements. As a case study, this dissertation illustrates the application of
  individual abstraction methods to such systems. Furthermore, it demonstrates the
  application of collective abstraction methods at various accuracy levels to temporal
  behavior analysis of several genetic regulatory networks. This dissertation shows
  that analysis time of biologically relevant properties of such genetic regulatory
  networks can be improved from days of work to minutes of work using our methodology
  while maintaining reasonable accuracy.
publication: ''
---
